### ISSUE TITLE
Support interpolated strings and variables as redirect targets

### ISSUE THREAD

**Issue by kevsmith** [operable/cog#1358]
The following should all parse and work as expected (assuming `$room` is a valid variable):

`echo hello > $room`
`echo hello > '#${room}'`
`echo hello > "${room}"`

**Comment by kevsmith** [comment]
After several days of research I've decided to punt this feature to post 1.0 for the following reasons:

* Adding parser support for variable and interpolated strings as redirect targets is pretty straightforward and easy to do. Unfortunately, this is only the beginning.
* Allowing variables and interpolated strings as redirect targets changes Cog's redirect model from one where output targets are fixed during pipeline set up to one where output targets could conceivably change on every pipeline iteration. Changes here have a lot of potential ripple effects into all aspects of pipeline processing such as output routing, template processing, and redirect target validation.

Implementing the new output routing logic, testing it, and then ensuring no regressions were introduced seems like a tall order given that we'd like to ship Cog 1.0 soon. For these reasons I've decided to delay implementing this enhancement until after 1.0 ships.
